### PR TITLE
Refactoring: replace utils/backend with Vuex modules/backend

### PR DESCRIPTION
**Pull request by shapkarin**
Currently the propose of this PR is to achieve the SSR functionality for the rest of the app – at least to make `Tip page` and `Profile page` sharable with social networks.
This PR is the second part of that PRs: https://github.com/aeternity/superhero-ui/pull/951 https://github.com/aeternity/superhero-ui/pull/985

### CODE REVIEW

File: src/store/modules/aeternity.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import FUNGIBLE_TOKEN_CONTRACT from 'aeternity-fungible-token/FungibleTokenFullI
 import { BigNumber } from 'bignumber.js';
 import { IS_MOBILE_DEVICE } from '../../utils';
 
+// todo: getters
```
Comment (mradkov) [inline src/store/modules/aeternity.js:15]: Can you remove this comment?

If the PR is still work in progress you can mark it as draft.

File: src/store/modules/backend.js [hunk 2]
```diff
@@ -2,9 +2,9 @@
 import Vue from 'vue';
 import { times } from 'lodash-es';
 // eslint-disable-next-line import/no-cycle
-import Backend from '../../utils/backend';
 import { createDeepLinkUrl } from '../../utils';
 
+// todo: getters
```
Comment (mradkov) [inline src/store/modules/backend.js:7]: Can you remove this comment?

If the PR is still work in progress you can mark it as draft.

Comment (shapkarin) [reply src/store/modules/backend.js:7]: sure

Comment (shapkarin) [reply src/store/modules/backend.js:7]: no it's not WIP. but I will remove all useless comments.

Comment (shapkarin) [reply src/store/modules/backend.js:7]: I just mention that https://github.com/aeternity/superhero-ui/issues/992 in this comment

Comment (shapkarin) [reply src/store/modules/backend.js]: @mradkov done, all useless comments are removed. anyway I have an issues for the same propose.

**Review by mradkov — COMMENTED** [review]
Code looks good. @Liubov-crypto can you check if everything works as intended, thanks!

**Comment by shapkarin** [comment]
@Liubov-crypto please test this:
- load comments
- load tips list at the main page
- load tip at the tip page
- load all profile data and profile page itself
- auth with extension and web wallet
- add new comment
- if user is auth after closed page and reopen again
- user tip list at profile page
- user comments list at profile page
- pin to channel (menu at three dots at the tip)
- unpin item
- list of pins
- change profile data
- change profile image
- claim for the tip (menu at three dots at the tip)
- report post (menu at three dots at the tip)
- if profile images loads fine
- change profile biography
- if AE to fiat values are works fine
- if infinity scroll for tips list works fine for the all kind of tip lists
- if all comments loads fine
- if comment page loads
- if balance at profile works fine (actually if all profile data loads fine and can be changed)
- comments count for the tip
- create a tip with zero AE value

**Comment by Liubov-crypto** [comment]
My Profile:
- Change Profile Info - failed
- Change Avatar/Cover Pic - failed
- Avatar - not uploaded 
- Cookies policy  - failed (not active)
Feed: 
- Create a comment - failed
- Report Post - failed
- Claim - pass
- Pin to Channel - failed
- Searching on the Feed - failed
- Tip on the feed - pass
- Change currency on the right section - pass
- Search - failed
•	load comments - pass
	•	load tips list at the main page - pass
	•	load tip at the tip page - pass
	•	load all profile data and profile page itself - failed (no .chain name, no profile picture) 
	•	auth with extension and web wallet - pass/pass
	•	add new comment - failed
	•	if user is auth after closed page and reopen again - pass ???
	•	user tip list at profile page - pass
	•	user comments list at profile page - pass
	•	pin to channel (menu at three dots at the tip) - pass, but there was no Pop-up confirmed this transaction
	•	unpin item - failed
	•	list of pins - failed, I see all content from the sh feed in my list, although I not pinned this 
	•	change profile data - failed
	•	change profile image - failed
	•	claim for the tip (menu at three dots at the tip) - pass
	•	report post (menu at three dots at the tip) - failed
	•	if profile images loads fine - failed/ not loaded 
	•	change profile biography - failed
	•	if AE to fiat values are works fine - pass
	•	if infinity scroll for tips list works fine for the all kind of tip lists - pass
	•	if all comments loads fine - pass
	•	if comment page loads - pass
	•	if balance at profile works fine - pass(actually if all profile data loads fine and can be changed) - failed
	•	comments count for the tip - pass
	•	create a tip with zero AE value - pass
Voting: 
Create voting - pass
Vote - pass
Revoke vote - pass
Search - pass
Meet: 
Login with web wallet - pass
Login with extension - pass
League: 
Login with web wallet - pass
Login with extension - pass
FAQ - pass

Got this warning on Voting, but the poll was created successfully:

vot1

Also, there is an issue with displaying content images and .chain names:

feed

**Comment by shapkarin** [comment]
• change profile image - failed
I checked that right now. works well. it's just some UI bug in `develop` branch - remove avatar icon overlaps the upload icon.

**Comment by shapkarin** [comment]
@Liubov-crypto please also try check the same cases with web wallet and with extension.

**Comment by shapkarin** [comment]
@mradkov currently I can't find a reason why `this` is `undefined` at the `backendAuthMixin.js` after I confirm changes from the web wallet and redirected back to the profile page. This is the reason why @Liubov-crypto can't change a profile data. Need to investigate that tomorrow.

**Comment by shapkarin** [comment]
basically the reason why `this` is `undefined` is because `beforeRouteEnter`
```
    // called before the route that renders this component is confirmed.
    // does NOT have access to `this` component instance,
    // because it has not been created yet when this guard is called!
````
ref: https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards

**Comment by shapkarin** [comment]
@davidyuk can you have a look at the two last comments. Not sure how to refact it.

**Comment by shapkarin** [comment]
Currently here is the issue because context is undefined but not the component.
https://github.com/aeternity/superhero-ui/blob/ed5067f8d988bd36f95aeb545677f32aaa98d5a4/src/utils/backendAuthMixin.js#L9-L19

**Comment by shapkarin** [comment]
I think to use the same at the create method of each component. But will try to add the special Vuex action that also can parse the query params.

**Comment by shapkarin** [comment]
I diced to just replace the `beforeRouteEnter` in the mixin in that way 
https://github.com/aeternity/superhero-ui/pull/993/commits/01426b6635a660992b239fb2955490cfd6ba4ded#diff-04e3f012be652c314d038888ea6d0e27bec4b770f8bc21d5d9abc3f8d5c764c8

**Comment by shapkarin** [comment]
@davidyuk please, also check ~~this latest commit https://github.com/aeternity/superhero-ui/pull/993/commits/be675432fc69091eb53b8dbff8eedf2bff3add3f~~

this commit: https://github.com/aeternity/superhero-ui/pull/993/commits/3fb21b0465b4247617b2c038f1c794875570dad1

Looks like previous code in develop was wrong. The code was:
https://github.com/aeternity/superhero-ui/blob/dfdbe57bf98e655da53de5d24001448c6d99ac09/src/store/index.js#L45-L48
But the response share is an object and don't have `forEach` method:
Снимок экрана 2021-01-27 в 12 29 39

File: src/store/index.js [hunk 3]
```diff
@@ -36,15 +35,15 @@ export default new Vuex.Store({
...
-    async updatePinnedItems({ commit, state: { address } }) {
-      commit('setPinnedItems', await Backend.getPinnedItems(address));
+    async updatePinnedItems({ commit, dispatch, state: { address } }) {
+      commit('setPinnedItems', await dispatch('backend/getPinnedItems', address));
     },
     async updateCookiesConsent({ commit, dispatch }) {
-      dispatch('backend/callWithAuth', { method: 'getCookiesConsent' })
-        .then((list) => list.forEach(({ scope, status }) => commit('setCookiesConsent', { scope, status: status === 'ALLOWED' })));
+      const payload = await dispatch('backend/callWithAuth', { method: 'getCookiesConsent' });
+      commit('setCookiesConsent', { payload, status: true });
```
Comment (davidyuk) [inline src/store/index.js:46]: I agree that you can set all of them at once, but I don't get why `status` need then

File: src/components/Avatar.vue [hunk 4]
```diff
@@ -32,15 +32,10 @@ export default {
...
-      return `${Backend.getProfileImageUrl(this.address)}?${key || ''}`;
-    },
   }),
-  watch: {
-    profileImageUrl() {
-      this.error = false;
-    },
+  async created() {
+    const key = this.profile?.signature?.slice(0, 5);
+    this.profileImageUrl = `${await this.$store.dispatch('backend/getProfileImageUrl', this.address)}?${key || ''}`;
```
[the remark this answers is not in the document]
Comment (shapkarin) [reply src/components/Avatar.vue:38]: right, thanks a lot just forgot and it was the hotfix. I will!

File: src/components/layout/sendTip/SendPost.vue [hunk 5]
```diff
@@ -125,11 +124,8 @@ export default {
       }
     },
     async postWithZeroBalance(title, media) {
-      const signature = await this.$store.dispatch(
-        'aeternity/postWithoutTipSignature',
-        { title, media },
-      );
-      return Backend.sendPostWithoutTip({
+      const signature = await this.$store.dispatch('postWithoutTipSignature', { title, media });
```
[the remark this answers is not in the document]
Comment (shapkarin) [reply src/components/layout/sendTip/SendPost.vue:127]: this actions is a part of the aethernify module that actually namespaced.

File: src/components/tipRecords/TipPreview.vue [hunk 6]
```diff
@@ -207,6 +206,11 @@ export default {
       }].find(({ regex }) => regex.test(this.tip.url))?.component;
     },
   },
+  async created() {
+    this.tipPreviewImage = this.isPreviewToBeVisualized
+      && this.tip.preview.image !== null
+      ? await this.$store.dispatch('backend/getTipPreviewUrl', this.tip.preview.image) : '';
```
[the remark this answers is not in the document]
Comment (shapkarin) [reply src/components/tipRecords/TipPreview.vue:212]: thanks! that's much better approach to do that.

File: src/store/index.js (hunk 3 above)
Comment (shapkarin) [reply src/store/index.js:46]: me too, let's have a call.

**Comment by shapkarin** [comment]
@davidyuk thanks for the conversation at the call. I will draft this PR and will choose the other direction to achive the same goal - avoid to import store explicitly.

**Pull request closed without merging**